### PR TITLE
[CU-1rpc34a] pallet-assets benchmarks

### PR DESCRIPTION
**Pull request by Br1ght0ne**

### CODE REVIEW

**Comment by itsbobbyzzz** [comment]
Task linked: [CU-1rpc34a Write Benchmarks](https://app.clickup.com/t/1rpc34a)

**Review by hussein-aitlahcen — APPROVED** [review]

**Comment by hussein-aitlahcen** [comment]
Any chance we get this one rebased and merged?

**Comment by Br1ght0ne** [comment]
@hussein-aitlahcen Rebased, please review and we can merge.

**Comment by hussein-aitlahcen** [comment]
@Br1ght0ne It's new work not related to current work in the branch, can't you rebase on top of it?

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: nightly-2021-09-01
```
Comment (dzmitry-lahoda) [inline .github/workflows/check.yml:23]: not sure, i would prefer to retain specific nightly, even if it is old. either we risk to break ci at some point and also ci may be slower checking out new rust each day

File: frame/assets/src/lib.rs [hunk 2]
```diff
@@ -805,19 +993,19 @@ pub mod pallet {
...
-					return <<T as Config>::Currency>::balance_on_hold(who)
+					return <<T as Config>::NativeCurrency>::balance_on_hold(who)
 				}
 				<<T as Config>::MultiCurrency>::balance_on_hold(asset, who)
 			}
 
 			fn can_hold(asset: Self::AssetId, who: &T::AccountId, amount: Self::Balance) -> bool {
 				if asset == T::NativeAssetId::get() {
-					return <<T as Config>::Currency>::can_hold(who, amount)
+					return <<T as Config>::NativeCurrency>::can_hold(who, amount)
```
Comment (dzmitry-lahoda) [inline frame/assets/src/lib.rs:1008]: not sure about merge strategy here. are you goint to  merge previous PR before this?

File: frame/composable-traits/src/set_by_key.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pub trait SetByKey<Key, Value> {
+	fn set(k: Key, value: Value) -> Result<(), Value>;
```
Comment (dzmitry-lahoda) [inline frame/composable-traits/src/set_by_key.rs:2]: could drop one liner doc comment about?

**Comment by dzmitry-lahoda** [comment]
image
it is impossible to merge until all verified. i tried to times to do verify in the past via rebase and it was ugly

**Comment by hussein-aitlahcen** [comment]
@Br1ght0ne this PR is subsumed by #293 right? If so, can we close it?

**Pull request closed without merging**